### PR TITLE
chore: updates config and templates for node 18 compatibility

### PR DESCRIPTION
**Pull request by krpeacock**
# Description

Updates to package.json and the webpack config for Node 18

* Updates webpack packages
* changelog

Supports [SDK-422]

# How Has This Been Tested?

Manually tested - 

* nvm install 18
* nvm use 18
* dfx new
* dfx start --background
* dfx deploy
* npm start
* verify that calls from the frontend go through

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

[SDK-422]: https://dfinity.atlassian.net/browse/SDK-422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -122,6 +122,14 @@ This incorporates the following executed proposals:
 
 Added the ic-btc-adapter binary to the cache.
 
+= 0.9.4
+
+== DFX
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:127]: There's already a section for dfx in the new version, which will be dfx 0.10.0

Comment (krpeacock) [reply CHANGELOG.adoc]: I see now - updated with the correct spot

Comment (ericswanson-dfinity) [reply CHANGELOG.adoc]: It's still kind of in an odd place - `===` under `== Replica`

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -122,6 +122,10 @@ This incorporates the following executed proposals:
 
 Added the ic-btc-adapter binary to the cache.
 
+=== chore: updates starter project for Node 18
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:125]: Would you please move this up, under the `== dfx` section, rather than here under the `== replica` section?

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**